src/java/org/apache/cassandra/index/sai/virtual/SegmentsSystemView.java

```diff
@@ -0,0 +1,125 @@
+    public static final String MIN_SSTABLE_ROW_ID = "min_sstable_row_id";
+    public static final String MAX_SSTABLE_ROW_ID = "max_sstable_row_id";
+    public static final String START_TOKEN = "start_token";
+    public static final String END_TOKEN = "end_token";
+    public static final String MIN_TERM = "min_term";
+    public static final String MAX_TERM = "max_term";
+    public static final String COMPONENT_METADATA = "component_metadata";
+
+    public SegmentsSystemView(String keyspace)
+    {
+        super(TableMetadata.builder(keyspace, NAME)
+                           .partitioner(new LocalPartitioner(UTF8Type.instance))
+                           .comment("SSTable index segment metadata")
+                           .kind(TableMetadata.Kind.VIRTUAL)
+                           .addPartitionKeyColumn(KEYSPACE_NAME, UTF8Type.instance)
+                           .addClusteringColumn(INDEX_NAME, UTF8Type.instance)
+                           .addClusteringColumn(SSTABLE_NAME, UTF8Type.instance)
+                           .addClusteringColumn(SEGMENT_ROW_ID_OFFSET, LongType.instance)
+                           .addRegularColumn(TABLE_NAME, UTF8Type.instance)
+                           .addRegularColumn(COLUMN_NAME, UTF8Type.instance)
+                           .addRegularColumn(CELL_COUNT, LongType.instance)
+                           .addRegularColumn(MIN_SSTABLE_ROW_ID, LongType.instance)
+                           .addRegularColumn(MAX_SSTABLE_ROW_ID, LongType.instance)
+                           .addRegularColumn(START_TOKEN, UTF8Type.instance)
+                           .addRegularColumn(END_TOKEN, UTF8Type.instance)
+                           .addRegularColumn(MIN_TERM, UTF8Type.instance)
+                           .addRegularColumn(MAX_TERM, UTF8Type.instance)
+                           .addRegularColumn(COMPONENT_METADATA,
+                                             MapType.getInstance(UTF8Type.instance,
+                                                                 MapType.getInstance(UTF8Type.instance, UTF8Type.instance, false),
+                                                                 false))
+                           .build());
+    }
+
+    @Override
+    public DataSet data()
+    {
+        SimpleDataSet dataset = new SimpleDataSet(metadata());
+
+        forEachIndex(indexContext -> {
+            for (SSTableIndex sstableIndex : indexContext.getView())
+            {
+                sstableIndex.populateSegmentView(dataset);
+            }
+        });
+
+        return dataset;
+    }
+
+    private void forEachIndex(Consumer<IndexContext> process)
+    {
+        for (KeyspaceMetadata ks : Schema.instance.getUserKeyspaces())
+        {
+            Keyspace keyspace = Schema.instance.getKeyspaceInstance(ks.name);
+            if (keyspace == null)
+                throw new IllegalArgumentException("Unknown keyspace " + ks.name);
```

Comment:
nit: `IllegalStateException`

I guess this could happen if a keyspace is being dropped? Might be worth the message indicating that's possible...